src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -5157,7 +5157,9 @@ void StorageReplicatedMergeTree::readParallelReplicasImpl(
 {
     auto table_id = getStorageID();
 
-    auto parallel_replicas_cluster = local_context->getCluster(local_context->getSettingsRef().cluster_for_parallel_replicas);
+    auto scalars = local_context->hasQueryContext() ? local_context->getQueryContext()->getScalars() : Scalars{};
```

Comment:
is there a reason we get scalars here?

Reply:
It's leftovers. Will remove it. Thanks

Reply:
@rschu1ze Shouldn't it be caught during our build? (with `-Wunused-variable`)

Reply:
It should and it does. Unfortunately, the compiler only complains about simple cases like `auto scalars = Scalars{};`.  Whenever functions with potential side effects (like `hasQueryContext()` or `getScalars()`) are involved, the warning doesn't trigger.